src/os/bluestore/BlueStore.h

```diff
@@ -194,7 +200,7 @@ class BlueStore : public ObjectStore {
 
     Collection(BlueStore *ns, coll_t c);
   };
-  typedef ceph::shared_ptr<Collection> CollectionRef;
+  typedef boost::intrusive_ptr<Collection> CollectionRef;
```

Comment:
For bluestore, open_collection will return a intrusive_ptr from coll_map instead of "new structure"?